### ISSUE TITLE
[Windows] client crashs every after few MBs upload on large data

### ISSUE THREAD

**Issue by adeel109** [owncloud/client#4975]
### Expected behaviour
Hi,
I am running OwnCloud from a year now and it works perfectly.

recently i have build another server with latest OwnCloud 9 version with Nginx and MariaDB and on different sites i have deployed it to sync data. On 5 sites its working perfectly it means server itself is fine but on one site we have few 80GB folders and one folder carries around 15,000 to 20,000 files mostly pictures. 

when i start sync client, it crash and  shows popup message Windows Error Reporting nothing else. the first folder was all synced after repeated few crash but on second folder it crash and crash every after few MBs of data. if I skip that folder where it crash then it does on next or next coming sub-folder and start crash again where it likes.

![image](https://cloud.githubusercontent.com/assets/19977655/16122891/decee99e-33e0-11e6-9800-052c0de49970.png)

The crashing was so annoying so i write a script which run every after 5 mintues to terminate owncloud service and run client again. it helped a little but it kills and data loss can happen. 

Originaly data was placed on Windows Server 2008 and i though it might be OS issue then i shifted all the data to another Server carries Server2012 but no luck and it does the same.

if anyone have experience on syncing large data of pictures on OwnCloud might have Ideas.

any guidance would be very much appreciated as the problem leads OwnCloud to unreliability and we would not be sure that the sync is running all the time.
I dont know where to get the logs of it. if anyone needs please let me know where to get that.

Thanks
Adi

**Comment by MU7L3Y** [comment]
You need to fill in the template with your setup information. Simple one - what client are you using? Try the latest 2.2.1

**Comment by guruz** [comment]
Would you please try a nightly build? We have fixed things there which might help:
http://download.owncloud.com/desktop/daily/ownCloud-2.2.1.6173-nightly20160617-setup.exe

Please report back then...

**Comment by guruz** [comment]
Can you also watch the memory usage?

**Comment by adeel109** [comment]
Thanks for the fix replys. currently i am using 2.2.0 Client. and now going to change the version Guruz gave which is Owncloud 2.2.1.  
i have 31GB Ram on it and i will check the usage as well and will come back after .

Thanks Again.

**Comment by dragotin** [comment]
Yes, 2.2.0 was known to crash in certain situations, that is fixed in 2.2.1. Thanks.

**Comment by adeel109** [comment]
Brilliant. it looks Fixed and it has synced 3.8GB data and haven't been crashed .

Thanks a lot.

**Comment by OlivierS1** [comment]
I have been experiencing this issue with client 2.2.2 and server 9.0.2

**Comment by guruz** [comment]
@OlivierS1 Could you please try a 2.2.3 nightly build and report if this fixes it? http://download.owncloud.com/desktop/daily/testpilotcloud-2.2.3.6229-nightly20160706-setup.exe

**Comment by OlivierS1** [comment]
Hi Guruz,

I have installed the nightly file you pointed out and I think it resolved the issue.
Just to provide a bit more info on the background, I have one owncloud account hosting a little over 650GB of data spanning over 750,000 files.
I am running simultaneous update from my main PC and my laptop and both seem to proceed without any issues. I did delete a few files since installing the nightly built, but I doubt they would have a major impact on the outcome.
I will need to experiment further, but it is not a trivial task as owncloud client seem to go over every single files stored on the server before proceeding with updating my PC or laptop, eventhough I am restoring to a clean new directory.
I would say, it took in excess of 3 hours from the client to start downloading the files,
I wish downloading would be performed right away, but this can be a question for another thread.
As an extra note, when both laptop and PC started to download the files from owncloud account, I did try to recover the deleted files from within the owncloud webbrowser. The outcome was not pleasant as the web interface froze for a while and communication between PC and laptop were terminated.
In a nutshell, clients needs to go over the 3 hours of check again before downloading the files.
I am going to leave both PC and laptop running overnight, both running version 2.2.2 and 2.2.3 release of owncloud client. Each one assign its own folder location. I hope there isn't going to be any contentions.
I will try to report back when I get more insight into the issues.

**Comment by OlivierS1** [comment]
Version 2.2.3 nightly has not resolved the issue.
I had both 2.2.2 and 2.2.3 nightly running overnight, and in the morning my desktop had the Runtime library window error. 
The laptop was still running both versions, but it only downloaded less than 7GB of data made up of around 20,000 files. To be on the safe, side, I closed 2.2.2 on the laptop this morning and let follow its course all day to find it also stopped with the same error message.

**Comment by guruz** [comment]
@OlivierS1 Sounds not good.
I wonder if you hit a memory leak? Does the client memory usage in task manager grow infinitely?
Your amount of files is huge.. maybe it's better to have several sync folders configured with sub folders of your data.

**Comment by OlivierS1** [comment]
I don't think it is memory leak related but I can't say for sure at the time the error message came up.
When the error messages and the client were closed, the memory wasn't maxed out and would only drop by about 500MB.
I noticed owncloud client was still keeping network active, similar to when it checks for new files.
I am in the process of centralizing my data from several computers into a single account, which caused a large amount of redundant files. Owncloud should help leverage this issue. 
The size and the number of files will go down for this account once I have sorted out redundant files, but eventually it will keep on growing.

**Comment by adeel109** [comment]
@OlivierS1. making sub-folders for sync helps a lot and it takes less time on restarting sinc and if sync fails on specific folder it tells you where it crashes so you can explore in it for cause. I dont know the internal data structure but i think making huge folders: specific datastructure tree might make problem in db for organizing it and might cause issues in huge delays or crash as it goes beyand its handling capacity, Owncloud team might tell better about it.

i think you might have a write protected folders or files as well where Client Sync doesnt have full permissions. like some Apple Mac file formats would not give permissions for read write where it tries to sync but it cannot and it timed out while multiple tries etc.

**Comment by guruz** [comment]
2.3.0 beta1 has HUGE memory IMPROVEMENTS when syncing a lot of files.
https://owncloud.org/changelog/desktop/
